src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,66 @@
+package gift.controller;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import gift.Model.Product;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+    private final Map<Long, Product> products = new HashMap<>();
+
+    @GetMapping("")
+    public List<Product> getAllProducts(){
+        return new ArrayList<>(products.values());
+    }
+
+    @GetMapping("/{id}")
+    public Product getProduct(@PathVariable("id") Long id){
+        if(!products.containsKey(id)){
+            System.out.println("Product가 존재하지 않습니다.");
+            return null;
+        }
+        return products.get(id);
+    }
+
+    @PostMapping()
+    public void addProduct(@RequestBody Product product){
+        if(products.containsKey(product.id())){
+            System.out.println("이미 Product가 존재합니다.");
+            return;
+        }
+        products.put(product.id(), product);
+        System.out.println("추가됨");
+    }
+
+    @PutMapping("/{id}")
+    public void updateProduct(@RequestBody Product product, @PathVariable("id") Long id){
+        if(!products.containsKey(product.id())){
+            System.out.println("{%d}에 매칭되는 Product가 이미 존재하지 않습니다.".formatted(id));
```

Comment:
System.out.printf()를 사용하는 방법도 있습니다. 참고만 해주세요~
```suggestion
            System.out.printf("{%d}에 매칭되는 Product가 이미 존재하지 않습니다.%n", id);
```  


한 가지 더 팁을 드리면 System.out.println()을 사용해서 출력하는 형태는 안티 패턴인데요,  
만약 처음 접하는 개념이라면 무시해도 좋습니다 :)   

Logger를 사용해서 로그를 출력하면 기본 포맷(날짜, 시간 등 필요한 추가 정보)부터 로그 레벨 제어, 성능 향상을 얻을 수 있습니다.  


```java
    private static final Logger log = LoggerFactory.getLogger(ProductController.class);

    // 생략
    public ResponseEntity<Product> getProductById(@PathVariable Long id) {
           // ...
          log.info("hello world!");
    }
```

<img width="1508" alt="image" src="https://github.com/kakao-tech-campus-2nd-step2/spring-gift-product/assets/28697758/b98d7873-a6b5-4827-bd08-9b571afa4761">